### DISCUSSION TITLE
Set VirtWifi DNS address to 8.8.8.8 even after reboot?

### DISCUSSION THREAD

**Discussion by hyunki** [LSPosed/MagiskOnWSALocal#197] [Q&A] [answered]
Hi,
WSA keep forget VirtWifi setting after rebooting windows11.
Reddit suggest that WSA will follow PC's network adaptor setting( or Hyper-V which is disabled for me).
Since I have two network adaptor so I disabled wifi adaptor. Currently only Ethernet is working.
![image](https://user-images.githubusercontent.com/25163778/196640822-edb5f8cf-ce4d-4e23-87d7-9d1b690d3d0e.png)

I don't understand where these value are come from.
None of them are match my pc network settings.
I have to manually change DNS address of VirtWifi to 8.8.8.8 whenever I want to use internet in WSA.

If I enable "Advanced networking" option on, WSA will connected to "Ethernet" not "VirtWifi"
I don't have to change DNS to connect Internet. but when I do this, I can't use FTP.
FTP related apps are auto closed.

**Comment by Howard20181** [comment, marked as answer]
This is by design
If you don't turn on Advanced Networking, Windows will enable network address translation and the address in the WLAN settings will be the address of Hyper-V's NAT adapter
If you need to make it consistent with Windows please turn on Advanced Networking.

**Reply by hyunki** [reply]
Thank you for the reply.
WSA establish Ethernet connection when advanced network on.
In this state, I can't use ftp. It's not about the connection problem.
Apps just close without any notification. I don't know why. 
CX files, Turbo Ftp, FTP Server, none of them are working.
To transfer file, ADB push pull is the only answer?
It will be really helpful if u could give me a some hint.
Thanks again.

**Reply by Howard20181** [reply]
For transferring from WSA to Windows, you can use WSA's built-in File App (which can be opened from WSA settings), long press a file, select Open Method from the 3-dot menu in the upper right corner, then select Windows Default Preferences, and the file will be saved to the Windows user's Downloads folder.
For transferring from Windows to WSA, you can use [SMB client](https://github.com/google/samba-documents-provider) to mount the shared file check on Windows to WSA.

**Reply by williamvyong** [reply]
mayb u can try kde connect 
by entering back ur own ip with advanced network turn on

**Reply by hyunki** [reply]
> mayb u can try kde connect by entering back ur own ip with advanced network turn on

Thank you for letting me know such a good program.
I'll test it. Microsoft store downloading takes a million years..

**Reply by hyunki** [reply]
> mayb u can try kde connect by entering back ur own ip with advanced network turn on

Window version take time to install(downloading stops) but WSA to tablet, smartphone, laptop works fine. Thank you very much!
![image](https://user-images.githubusercontent.com/25163778/197397287-05ac9eb6-fd85-4bac-9e17-b7f56aeed0bb.png)
It tooks forever...
